src/approve.ts

```diff
@@ -3,27 +3,34 @@ import * as github from "@actions/github";
 import { RequestError } from "@octokit/request-error";
 import { Context } from "@actions/github/lib/context";
 
-export async function approve(token: string, context: Context) {
-  const { pull_request: pr } = context.payload;
-  if (!pr) {
+export async function approve(
+  token: string,
+  context: Context,
+  pr_number: number
+) {
+  if (!pr_number) {
+    pr_number = context?.payload?.pull_request?.number || 0;
```

Comment:
By making `prNumber` optionally `undefined`, we can remove the `|| 0`.

I also think we can skip some of those optional chaining operators as `context` shouldn't be `undefined`, and [neither should `payload`](https://github.com/actions/toolkit/blob/4bf916289e5e32bb7d1bd7f21842c3afeab3b25a/packages/github/src/context.ts#L10). [`pull_request`](https://github.com/actions/toolkit/blob/4bf916289e5e32bb7d1bd7f21842c3afeab3b25a/packages/github/src/interfaces.ts#L24) can be undefined so we should leave that one in.

```suggestion
    prNumber = context.payload.pull_request?.number;
```

Reply:
no problem!